packages/@aws-cdk/cdk/lib/app.ts

```diff
@@ -10,13 +10,16 @@ const APP_SYMBOL = Symbol.for('@aws-cdk/cdk.App');
  */
 export interface AppProps {
   /**
-   * Automatically call run before the application exits
+   * Automatically call `synth()` before the program exits.
    *
-   * If you set this, you don't have to call `synth()` anymore.
+   * If you set this, you don't have to call `synth()` explicitly. Note that
+   * this feature is only available for certain programming languages, and
```

Comment:
which languages are supported / which ones are not?

Reply:
I am not 100% sure TBH. I think some jsii languages don't implement the exit hook properly and I wanted to set expectations that this is not 100% working... We need to fix it obvsiouly and then remove this warning.